functions/components/organization/store.js

```diff
@@ -78,7 +71,41 @@ export const getOrganizationById = (id, clean=false) => {
         //     if (err) reject(err)
 
         //     resolve(doc)
-        // })
+        // })*/
+
+        orgModel.aggregate([
+            { $match: { _id:mongoose.Types.ObjectId(id) }},
+            { 
+                $lookup: {
+                    from: "clients",
+                    localField: "owner",
+                    foreignField: "_id",
+                    as: "admin",
+                }
+            },
+            {
+                $addFields: {
+                    admin: { $arrayElemAt: ["$admin", 0] }
+                }
+            },
+            {
+                $project: {
+                    "admin.password": 0,
+                    "admin.passphrase": 0
+                }
+            }
+        ])
+        .then((result) => {
+            if (result.length === 0) {
+                reject("Organization not found");
+            } else {
+                resolve(result[0]);
```

Comment:
Here we need to hydrate the response if the function is not receiving the clean option, because the object returned is a plain javascript object, and there are other controllers that are using this function to query the organization, and perform operations over it. For example is blocking the product creation because the method save() doesnt exist for plain objects.

This problem can be solved with the following line of code:

orgModel.hydrate(result[0]) and return this if the consumer of the function has not passed the clean parameter